java/runtime/src/main/java/org/ray/runtime/RayNativeRuntime.java

```diff
@@ -59,15 +59,24 @@
     } catch (IOException e) {
       throw new RuntimeException("Couldn't load native libraries.", e);
     }
-    nativeSetup(RayConfig.create().logDir);
-    Runtime.getRuntime().addShutdownHook(new Thread(RayNativeRuntime::nativeShutdownHook));
-  }
 
-  public RayNativeRuntime(RayConfig rayConfig) {
-    super(rayConfig);
+    RayConfig globalRayConfig = RayConfig.create();
+    File logDirFile = new File(globalRayConfig.logDir);
+    if (!logDirFile.exists()) {
+      //noinspection ResultOfMethodCallIgnored
+      logDirFile.mkdirs();
+    }
+    nativeSetup(globalRayConfig.logDir);
+    Runtime.getRuntime().addShutdownHook(new Thread(RayNativeRuntime::nativeShutdownHook));
   }
 
-  protected void resetLibraryPath() {
+  /**
+   * The synchronized keyword is applied together with the static keyword. That ensures
+   * thread-safety thread-safety when creating multiple RayNativeRuntime instances concurrently.
+   * Anyway, the {@link RayConfig#libraryPath} field is unlikely to change between runtime
+   * instances.
+   */
+  private static synchronized void resetLibraryPath(RayConfig rayConfig) {
```

Comment:
What about moving this into `RayMultiWorkerNativeRutime`. So it only run once per `Ray.init`